mempool/clist_mempool.go

```diff
@@ -729,39 +799,30 @@ func (rc *recheck) setDone() {
 	rc.isRechecking.Store(false)
 }
 
-// setNextEntry sets cursor to the next entry in the list. If there is no next, cursor will be nil.
-func (rc *recheck) setNextEntry() {
-	rc.cursor = rc.cursor.Next()
-}
-
 // tryFinish will check if the cursor is at the end of the list and notify the channel that
 // rechecking has finished. It returns true iff it's done rechecking.
 func (rc *recheck) tryFinish() bool {
-	if rc.cursor == rc.end {
+	if rc.cursor == nil || rc.cursor == rc.end {
 		// Reached end of the list without finding a matching tx.
 		rc.setDone()
 	}
 	if rc.done() {
-		// Notify that recheck has finished.
-		select {
-		case rc.doneCh <- struct{}{}:
-		default:
-		}
+		close(rc.doneCh) // notify channel that recheck has finished
```

Comment:
Are these changes because of Lanes, or are they also applicable to `v1.x`?

Reply:
I think this was done because we now wait for every lane then close a channel and open a new one. So it is only because of lanes, not 100% sure that it will stay like this atm.

Reply:
I think this applies also to the `main` branch. I will create a separate PR with `main` as target.